### PR TITLE
Focus on appfile in docs part 3 (w/ ref docs)

### PR DESCRIPTION
**Pull request by resouer**
As the title said, ref: https://github.com/oam-dev/kubevela/issues/564

Fixes: https://github.com/oam-dev/kubevela/issues/568

### CODE REVIEW

File: docs/en/developers/references/devex/cli.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # KubeVela CLI
 
-Learn about general configurations for `vela` command line tool. For the usage of the CLI, please check the KubeVela's [user documentation](../../README.md) instead.
+The fully specification of KubeVela's command line is automatically generated in this [directory](../../../cli).
```
Comment (hongchaodeng) [inline docs/en/developers/references/devex/cli.md:3]:
```suggestion
The fully specification of KubeVela's command line is automatically generated in the [cli directory](../../../cli).
```

Comment (hongchaodeng) [reply docs/en/developers/references/devex/cli.md:3]: I think relative directory path won't work in website. Need to reference to https link.

File: docs/en/developers/references/traits/route.md [hunk 2]
```diff
@@ -1,5 +1,27 @@
...
+  my-service-name:
+    ...
+    route:
+      domain: example.com
+      issuer: tls
+      rules:
+        - path: /testapp
+          rewriteTarget: /
+```
+
```
Comment (zzxwill) [inline docs/en/developers/references/traits/route.md:24]: Great, that's the nice reference for developers to write appfile at it's said in #577 :)

**Comment by resouer** [comment]
@hongchaodeng @wonderflow rebased, please review :-)

File: docs/_sidebar.md [hunk 3]
```diff
@@ -28,9 +28,19 @@
...
+  - [Appfile](/en/developers/references/devex/appfile.md)
+  - Capabilities
+    - Workload Types
+      - [Web Service](/en/developers/references/workload-types/web-service.md)
+      - [Task](/en/developers/references/workload-types/task.md)
+      - [Worker](/en/developers/references/workload-types/backend-worker.md)
+    - Traits
+      - [Route](/en/developers/references/traits/route.md)
+      - [Autoscale](/en/developers/references/traits/autoscale.md)
+      - [Rollout](en/developers/references/traits/rollout.md)
```
Comment (wonderflow) [inline docs/_sidebar.md:42]:
```suggestion
      - [Rollout](/en/developers/references/traits/rollout.md)
```

File: docs/_sidebar.md [hunk 4]
```diff
@@ -28,9 +28,19 @@
...
+      - [Metrics](en/developers/references/traits/metrics.md)
```
Comment (wonderflow) [inline docs/_sidebar.md:43]:
```suggestion
      - [Metrics](/en/developers/references/traits/metrics.md)
```

File: docs/en/developers/learn-appfile.md [hunk 5]
```diff
@@ -146,39 +147,33 @@ spec:
...
-
-> Below is a simplified [k8s job example](https://kubernetes.io/docs/concepts/workloads/controllers/job/#running-an-example-job) using Vela:
+By now we have deployed a *[Web Service](references/workload-types/webservice.md)*. We can also add another service of *[Task](references/workload-types/task.md)* type in the same app:
 
 ```yaml
 services:
   pi:
+    type: task
     image: perl 
     cmd: ["perl",  "-Mbignum=bpi", "-wle", "print bpi(2000)"]
```
Comment (wonderflow) [inline docs/en/developers/learn-appfile.md:157]: does it really work?

**Review by wonderflow — APPROVED** [review]

File: docs/en/developers/learn-appfile.md (hunk 5 above)
Comment (resouer) [reply docs/en/developers/learn-appfile.md:157]: I think it's copied from here https://dockerlabs.collabnix.com/kubernetes/cheatsheets/kubectl.html

```
# Start the perl container to compute π to 2000 places and print it out.
kubectl run pi --image=perl --restart=OnFailure -- perl -Mbignum=bpi -wle 'print bpi(2000)'
```

**Pull request merged**